### PR TITLE
Refactor Mergeability test cases

### LINKED ISSUE #3334: Refactor mergeability test cases
# Overview

Create function for each constructor of `MergeabilityTestCase` so we can wrap the test run itself.
https://github.com/rchain/rchain/blob/5a9b6d4/casper/src/test/scala/coop/rchain/casper/batch1/MergeabilityRules.scala#L273-L854

Each case can be defined something like this in which case we can easily run it separately and use reporting of the context of failed to Scala test.
```scala
it should "!X !X" in mergeableCase(S0)(S0)(Nil)(S0.rstate ++ S0.rstate)
```

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview

This pr is trying to address https://github.com/rchain/rchain/issues/3334.

Currently the Mergeability test cases are very difficult to run. We can not run the test cases separately very easily and it prevents us to debug and test the senario easily which make efficiency slow.

This can help the developers to identify the error test cases quicklier and makes life easy 

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/batch1/JoinMergeability.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
+package coop.rchain.casper.batch1
+
+import org.scalatest.{FlatSpec, Inspectors, Matchers}
+
+class JoinMergeability extends FlatSpec with Matchers with Inspectors with MergeabilityRules {
+  it should "J S S" ignore ConflictingCase(S0)(S1)(J_)(J_.rstate ++ S0.rstate)(
```
Comment (zsluedem) [inline casper/src/test/scala/coop/rchain/casper/batch1/JoinMergeability.scala:6]: all of the joins cases are still ignored in this pr.

**Comment by zsluedem** [comment]
newest version result is perfectly right even with pervalidator vault 
[mergeBalances.csv](https://github.com/rchain/rchain/files/6574642/mergeBalances.csv)

**Review by nzpr — APPROVED** [review]
LGTM

**Review by tgrospic — APPROVED** [review]
LGTM

**Comment by zsluedem** [comment]
bors merge

**Comment by zsluedem** [comment]
bors merge

**Pull request merged**